internal/resolve/catalog.go

```diff
@@ -114,7 +114,7 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 		if len(resolvedBundles) != 0 {
 			// We've already found one or more package candidates
 			currentIsDeprecated := isDeprecated(thisBundle, thisDeprecation)
-			priorIsDeprecated := isDeprecated(*resolvedBundles[0], priorDeprecation) // Slice index doesn't matter; the whole slice is either deprecated or not
+			priorIsDeprecated := isDeprecated(*resolvedBundles[len(resolvedBundles)-1], priorDeprecation)
```

Comment:
Am I correct in understanding that the expectation is that if the previously installed package is deprecated we expect it to be at the end of the slice of resolved bundles?

How would this work if there are more than one deprecated resolved bundles and the previously installed bundle is of higher semver than other deprecated bundles?

Reply:
The package at the end of the slice isn't the currently installed package, it's just the package we got from the previous catalog we looked through. 

But regardless, in your scenario, if the user isn't filtering out packages of a lower version (`UpgradeConstraintPolicyIgnore=true`) and our resolver finds multiple packages that are also deprecated then we would have an ambiguous resolution i.e. we will not decide for the user which package in which catalog we will downgrade to.